### ISSUE TITLE
Struggling to get switch to work

### ISSUE THREAD

**Issue by ituri** [normen/homebridge-433-arduino#24]
Hi,

I have an 3 elro plugs which can be triggered with the corresponding remote. I'm also running the Arduino software on a D1 with a C1101 connected to it. When I push the button for the first switch on my remote I can see the following output in my log:
```[1/13/2021, 4:51:26 PM] [Arduino RC Switch Platform] {"type":"pollin","message":{"systemcode":11,"unitcode":1,"state":"on"}}
[1/13/2021, 4:51:26 PM] [Arduino RC Switch Platform] {"type":"kerui_D026","message":{"unitcode":696864,"state":"closed"}}
[1/13/2021, 4:51:26 PM] [Arduino RC Switch Platform] {"type":"ev1527","message":{"unitcode":700143,"state":"closed"}}
[1/13/2021, 4:51:26 PM] [Arduino RC Switch Platform] {"type":"elro_800_switch","message":{"systemcode":11,"unitcode":1,"state":"on"}}
[1/13/2021, 4:51:26 PM] [Arduino RC Switch Platform] {"type":"elro_800_contact","message":{"systemcode":11,"unitcode":1,"state":"opened"}}
[1/13/2021, 4:51:26 PM] [Arduino RC Switch Platform] {"type":"elro_400_switch","message":{"systemcode":5,"unitcode":15,"state":"on"}}
```

I have added the following lines to my homebridge config:
```
            "switches": [
                {
                    "name": "Lampe Fensterbank",
                    "hint": "Lampe Fensterbank",
                    "type": "elro_800_switch",
                    "message": {
                        "systemcode": "11",
                        "unitcode": "1"
                    }
                },
```
When I press the button on my physical remote, the plugin correctly registers the state of the button as "on". However, pressing the button inside homebridge doesn't cause the switch to turn off.  It does say `[1/13/2021, 5:16:56 PM] [Arduino RC Switch Platform] Sent off code for Lampe Fensterbank` but nothing happens. What could I do here?

**Comment by normen** [comment]
Only "id" and "unit" are accepted values in the config file, maybe you can use those instead. If they don't get recognized you might be able to control the lamps using a different type (one they don't report). E.g. I can use quigg_gt1000 for switches tha report quigg_gt9000, which by itself doesn't work.

If neither works you'd have to add "systemcode" and "unit" as recognized options to the code - really idk why pilight uses a bunch of different names for whats essentially the same thing.

**Comment by ituri** [comment]
I tried other switches but that didn't work. 

I'm not much of a coder myself but I'm really willing to get this to work. Maybe I could simply find & replace "id" with "systemcode" and "unit" with "unitcode" in the code. Would I need to do this both for the arduino code as well as for the plugin?
Or is there any chance you might be adding support for other value names to future versions of the plugin?

**Comment by normen** [comment]
If you only have those switches that might just work 😆 Include the quotes in the search-replace, that might narrow it down to the right things.

If it works you could maybe show me a diff of that, might help to actually implement it properly at some point.

**Comment by superhenne** [comment]
Not sure if this question is still relevant, but still: I also use the elro_800_switch successfully. I noticed a difference in my JSON config: The ids are configured as numbers without quotes.
```
{
            "name": "C Licht",
            "type": "elro_800_switch",
            "message": {
                "systemcode": 24,
                "unitcode": 4
            }
        },
```

**Comment by normen** [comment]
> Not sure if this question is still relevant, but still: I also use the elro_800_switch successfully. I noticed a difference in my JSON config: The ids are configured as numbers without quotes.

Good catch, didn't even realize that but yes, it should be set without quotes for numbers.
